lib/autoupdate/agent/updater.go

```diff
@@ -773,12 +743,25 @@ func (u *Updater) update(ctx context.Context, cfg *UpdateConfig, target Revision
 			}
 			return trace.Wrap(err, "failed to start new version %s of Teleport", target)
 		}
+		u.Log.InfoContext(ctx, "Target version successfully installed.", targetKey, target)
 
 		if r := cfg.Status.Active; r.Version != "" {
 			cfg.Status.Backup = toPtr(r)
 		}
 		cfg.Status.Active = target
 	} else {
+		err := u.ExecSetup(ctx, false)
+		if errors.Is(err, context.Canceled) {
+			return trace.Errorf("check canceled")
+		}
+		if err != nil {
+			// If sync fails, we may have left the host in a bad state, so we revert linking and re-Sync.
+			u.Log.ErrorContext(ctx, "Reverting symlinks due to invalid configuration.")
+			if ok := revertConfig(ctx); ok {
+				u.Log.WarnContext(ctx, "Teleport updater encountered a configuration error and successfully reverted the installation.")
+			}
```

Comment:
Can we also log if the revert failed?

Reply:
`revertConfig` is a local function that always logs when it returns false.